### ISSUE TITLE
WebSocket fragmentation issues

### ISSUE THREAD

**Issue by Scormave** [Benedicht/BestHTTP-Issues#145]
After upgrading BestHTTP to version 2.8.1 some strange issues started to happen with SignalR Core and WebSocket. When sending a big messages server just closes a connection with deserialization error. Btw, I'm using MessagePack protocol.
I did a lot of research and found that if a message is more than 32Kb, and is fragmented by WebSocket, there is high chance that server will fail and close a connection.
Interesting, that after downgrade to version 2.7.0 the issue is gone completely and messages of size more than 32Kb are correctly fragmented and processed by server.

STR: Send a message that is more than 32Kb with WebSocket transport using SignalR Core.
Affected version: 2.8.1
Not reproduced version: 2.7.0

**Comment by Scormave** [comment]
For now to get everything work, I've reverted WebSocket changes and left other code untouched.

**Comment by Benedicht** [comment]
Hmm, this is quite interesting. The websocket framing mechanism got rewritten in v2.8.0, but i'm not aware of any changes that would change the end result.
The server however has a [MaximumReceiveMessageSize](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.signalr.huboptions.maximumreceivemessagesize?view=aspnetcore-5.0) that set to 32KB by default.

It would be really good to know what's going on under the hoods. so could you

1. Downgrade the whole plugin to v2.7.0,
2. set the plugin's log level to verbose,
3. send those large messages,
4. save somewhere your Editor.log
5. upgrade to the latest (v2.8.1) version
6. repeat steps 2.-4.
7. send over **both** Editor.logs (here, over email - besthttp@gmail.com, or on [discord](https://discord.gg/yD9tXwQ))

This way I could compare the two logs and hopefully get some clue what's going on.

**Comment by Benedicht** [comment]
Forgot to elaborate on the logging part. You can add the following line somewhere in your startup code:
```cs
BestHTTP.HTTPManager.Logger.Level = BestHTTP.Logger.Loglevels.All;
```

You can read more about logging in the plugin here: https://benedicht.github.io/BestHTTP-Documentation/pages/best_http2/global_topics/Logging.html

**Comment by Scormave** [comment]
I'm aware of MaximumReceiveMessageSize and it was set to 10Mb. Also, there are ApplicationMaxBufferSize and TransportMaxBufferSize which also have been increased to 10Mb. And turned on EnableDetailedErrors to catch info about errors.
As well, I've tried SignalR with a simple console application, and everything went fine.
No problem, I will prepare log files.

**Comment by Benedicht** [comment]
Is it happens only with MessagePack?

**Comment by Scormave** [comment]
Sent you an email with collected log files.

**Comment by Scormave** [comment]
> > Unfortunately, I haven't tried Json, because it can be hard to implement it. But with MessagePack it happens for sure.
> 
> I'm not sure i get it, the plugin comes with a default json implementation that can be used:
> 
> ```cs
> hub = new HubConnection(new Uri("https://localhost:5001" + this._path), new JsonProtocol(new LitJsonEncoder()));
> ```

Yes, but when I tried to use it, there were exceptions from serialization related to Guid, Int32 etc...
I just didn't want to dive into it.

**Comment by Benedicht** [comment]
Understood, also if you are sending/receiving binary data, MessagePack is a better choice.
